clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -683,8 +685,8 @@ public void onSuccess(ClientResponse response, RequestFuture<Map<TopicPartition,
      * @param listOffsetResponse The response from the server.
      * @param future The future to be completed when the response returns. Note that any partition-level errors will
      *               generally fail the entire future result. The one exception is UNSUPPORTED_FOR_MESSAGE_FORMAT,
-     *               which indicates that the broker does not support the v1 message format and results in a null
-     *               being inserted into the resulting map.
+     *               which indicates that the broker does not support the v1 message format and causes a null value
```

Comment:
> Regarding the comment, do you think it's better to remove the part about null value altogether, or have something like "a null offset value is returned for the partition in this case"?

It seems like the actual behavior is to only include offsets that have legitimate values. So maybe we can say that?

>Also, do you suggest the parameter type change to be done as part of this PR? Thanks.

Yes, I was thinking that it would simplify the logic above (we can replace the loop with a `putAll` perhaps).